proto/serve.proto

```diff
@@ -0,0 +1,224 @@
+  // The token to list the next page of RayServices.
+  string next_page_token = 3;
+}
+
+message ListAllRayServicesRequest {
+  // A page token to request the next page of results. The token is acquried
+  // from the nextPageToken field of the response from the previous
+  // ListRayServices call or can be omitted when fetching the first page.
+  string page_token = 2;
+  // The number of RayServices to be listed per page. If there are more
+  // RayServices than this number, the response message will contain a
+  // nextPageToken field you can use to fetch the next page.
+  int32 page_size = 3;
+}
+
+message ListAllRayServicesResponse {
+  repeated RayService services = 1; 
+  // The total number of RayServices for the given query.
+  int32 total_size = 2;
+  // The token to list the next page of RayServices.
+  string next_page_token = 3;
+}
+
+message DeleteRayServiceRequest {
+  // The name of the ray service to be deleted.
+  string name = 1;
+  // The namespace of the ray service to be deleted.
+  string namespace = 2;
+}
+
+message RayService {
+  // Required input field. Unique ray service name provided by user.
+  string name = 1;
+  // Required input field. ray service namespace provided by user
+  string namespace = 2;
+  // Required field. This field indicates the user who owns the ray service.
+  string user = 3;
+  // The config for ray serve deployments.
+  ServeDeploymentGraphSpec serve_deployment_graph_spec = 4;
+  // The cluster template.
+  ClusterSpec cluster_spec = 5;
+  // Output. The status for the ray service.
+  RayServiceStatus ray_service_status = 6;
+  // Output. The time that the ray service created.
+  google.protobuf.Timestamp created_at = 7;
+  // Output. The time that the ray service deleted.     
+  google.protobuf.Timestamp delete_at = 8;
+}
+
+message ServeDeploymentGraphSpec {
+    // Required input field. Import path for load the ray serve.
+    string import_path = 1;
+    // runtime env for loading the ray serve.
+    string runtime_env = 2;
+    // deployments config of the ray serve.
+    repeated ServeConfig serve_configs = 3;
+}
+
+message ServeConfig {
+  // Required input field. Unique ray service deployment name provided by user.
+  string deployment_name = 1;
+  // ray service deployment replicas, default set to 1.  
+  int32 replicas = 2;
+  // ray service deployment route prefix indicated by user.
+  string route_prefix = 3;
+  // max concurrent queries indicate for each ray service deployment.
+  int32 max_concurrent_queries = 4;
+  // user config to the ray serve deployment.
+  string user_config = 5;
+  // auto scaling config for ray serve deployment indicate by user.
+  string autoscaling_config = 6;
+  // actor options indicated resource of each deployment's actor.
+  ActorOptions actor_options = 7;
+}
+
+message ActorOptions {
+  string runtime_env = 1;
+  double cpus = 2;
+  double gpu = 3;
+  int32 memory = 4;
+  int32 object_store_memory = 5;
+  string resource = 6;
```

Comment:
what's the resource field for?

Reply:
Minor: This is actually the custom resource? I feel ray serve's definition is not that clear. As a user, I may be confused on cpu,memory,gpu and resources here. I can not figure out what to put. Do you have the same feeling? If we can use different name to clarify that, that would be great.

Reply:
yes, I've replace the name with `cpu_per_actor` ... and add more comments for this field.